skyportal/models/public_pages/public_source_page.py

```diff
@@ -26,23 +21,10 @@
 )
 
 
-def published_source_access_logic(cls, user_or_token):
-    """Return a query that filters PublicSourcePage instances based on user access."""
-    # if the user is a system admin, he can update and delete all published sources
-    # otherwise, he can only delete and update the published sources associate with his group
-    user_id = UserAccessControl.user_id_from_user_or_token(user_or_token)
-    query = sa.select(cls)
-    if not user_or_token.is_system_admin:
-        query = query.join(Source, cls.source_id == Source.obj_id)
-        query = query.join(GroupUser, Source.group_id == GroupUser.group_id)
-        query = query.filter(GroupUser.user_id == user_id)
-    return query
-
-
 class PublicSourcePage(Base):
     """Public page of a source on a given date."""
 
-    update = delete = CustomUserAccessControl(published_source_access_logic)
+    update = delete = accessible_by_groups_members
```

Comment:
that doesn't cause any issue for public sources not on a group? Just checking, I don't remember what baselayer does in that case.

Reply:
It was a mistake to change from CustomUserAccessControl to accessible_by_groups_members.